core/src/main/scala/org/apache/spark/internal/io/HadoopMapReduceCommitProtocol.scala

```diff
@@ -125,22 +126,27 @@ class HadoopMapReduceCommitProtocol(
       taskContext: TaskAttemptContext, dir: Option[String], spec: FileNameSpec): String = {
     val filename = getFilename(taskContext, spec)
 
-    val stagingDir: Path = committer match {
-      // For FileOutputCommitter it has its own staging path called "work path".
-      case f: FileOutputCommitter =>
-        if (dynamicPartitionOverwrite) {
-          assert(dir.isDefined,
-            "The dataset to be written must be partitioned when dynamicPartitionOverwrite is true.")
-          partitionPaths += dir.get
-        }
-        new Path(Option(f.getWorkPath).map(_.toString).getOrElse(path))
-      case _ => new Path(path)
-    }
+    if (forceUseStagingDir && !dynamicPartitionOverwrite) {
```

Comment:
when `spark.sql.hive.convertMetastoreParquet` or `spark.sql.hive.convertMetastoreOrc` is `false`, mean use hive serde. We also call `newTaskTempFileAbsPath`. Here will trigger rename. I suspect this is a conflict with hive serde logic.